src/components/Root/__tests__/LegacyRouteRedirection.test.tsx

```diff
@@ -0,0 +1,31 @@
+import { render } from '@testing-library/react'
+import { ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
+import { history } from 'src/routes/routes'
+import { PUBLIC_URL } from 'src/utils/constants'
+import LegacyRouteRedirection from '../LegacyRouteRedirection'
+
+describe('LegacyRouteRedirection', () => {
+  it('redirects legacy safe links and prepends the short chain name', () => {
+    history.push(`/#/safes/${ZERO_ADDRESS}`)
+
+    render(<LegacyRouteRedirection />)
+
+    expect(window.location.pathname).toBe(`/rin:${ZERO_ADDRESS}`)
+  })
+  it('redirects legacy load links and prepends the short chain name', () => {
+    history.push(`/#/load/${ZERO_ADDRESS}`)
```

Comment:
Could you add another test case with a URL like `/#/welcome`? I.e. with no address.